packages/api/cms-api/src/action-logs/action-logs.subscriber.ts

```diff
@@ -0,0 +1,83 @@
+import { AnyEntity, ChangeSet, ChangeSetType, EntityManager, EventSubscriber, FlushEventArgs, PostgreSqlDriver, raw } from "@mikro-orm/postgresql";
+import { Injectable } from "@nestjs/common";
+import { ModuleRef } from "@nestjs/core";
+
+import { SCOPED_ENTITY_METADATA_KEY, ScopedEntityMeta } from "../user-permissions/decorators/scoped-entity.decorator";
+import { ContentScope } from "../user-permissions/interfaces/content-scope.interface";
+import { ACTION_LOGS_METADATA_KEY, ActionLogMetadata } from "./action-logs.decorator";
+import { ActionLogsService } from "./action-logs.service";
+import { ActionLog } from "./entities/action-log.entity";
+
+@Injectable()
+export class ActionLogsSubscriber implements EventSubscriber {
+    constructor(
+        private readonly entityManager: EntityManager<PostgreSqlDriver>,
+        private readonly service: ActionLogsService,
+        private readonly moduleRef: ModuleRef,
+    ) {
+        entityManager.getEventManager().registerSubscriber(this);
+    }
+
+    async onFlush(args: FlushEventArgs): Promise<void> {
+        const changeSets = args.uow.getChangeSets();
+        for (const changeSet of changeSets.filter((changeSet) =>
+            Reflect.hasOwnMetadata(ACTION_LOGS_METADATA_KEY, changeSet.entity.constructor.prototype),
+        )) {
+            const entity = await this.createEntity(changeSet);
+            if (!entity) continue;
+            args.uow.computeChangeSet(entity, ChangeSetType.CREATE);
+            args.uow.recomputeSingleChangeSet(entity);
+        }
+    }
+
+    async createEntity(changeSet: ChangeSet<AnyEntity>): Promise<ActionLog | null> {
```

Comment:
Naming: createLog or createLogEntry. I was a bit confused why we'd need to create an entity ourselves.

Reply:
refactored with 410bcd03df9f52a479e14339572396fa32499ef1